docs/2022/puzzles/day18.md

```diff
@@ -6,6 +6,91 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 https://adventofcode.com/2022/day/18
 
-## Solutions from the community
+## Solution
 
-Share your solution to the Scala community by editing this page. (You can even write the whole article!)
\ No newline at end of file
+### Part 1
+
+The first part turns out to be fairly straight forward.  A cube has six sides, so if we count the total number of cubes and multiply this by six, we just need to subtract the number of connected sides.  We'll need to be able to check if two cubes are adjacent, so lets first define a function for that:
+
+```scala
+def adjacent(x: Int, y: Int, z: Int): Set[(Int, Int, Int)] = {
+  Set(
+    (x + 1, y, z),
+    (x - 1, y, z),
+    (x, y + 1, z),
+    (x, y - 1, z),
+    (x, y, z + 1),
+    (x, y, z - 1)
+  )
+}
+```
+
+Note since cubes are given to be 1⨉1⨉1, we can represent cubes as a single integral `(x, y, z)` coordinate which makes up the input for this function.  Then two cubes are adjacent (ie one of each of their sides touch) if and only if exactly one of their `(x, y, z)` components differ by one, and the rest by zero.
+
+Now given our cubes, we can implement our strategy in a fairly straight-forward manner with a `fold`:
+```scala
+def sides(cubes: Set[(Int, Int, Int)]): Int = {
+  cubes.foldLeft(0) { case (total, (x, y, z)) =>
+    val adj = adjacent(x, y, z)
+    val noAdj = adj.filter(cubes).size
+    total + 6 - noAdj
+  }
+}
+```
+We use a `Set` for `O(1)` membership lookups which we need to determine which adjacent spaces for a given cube contain other cubes.
```

Comment:
citing [the docs](https://docs.scala-lang.org/overviews/collections-2.13/performance-characteristics.html) im not sure we can say `O(1)` for immutable set (its listed as "effectively constant")
```suggestion
We use a `Set` for membership lookups which we need to determine which adjacent spaces for a given cube contain other cubes.
```

Reply:
I've removed the reference to `O(1)`, and said they're 'fast' with a link to the docs, since I wanted to provide some justification of using a set rather than say a list - what do you think of this?